internal/terraform/eval_import.go

```diff
@@ -72,6 +72,15 @@ func evaluateImportIdExpression(expr hcl.Expression, ctx EvalContext, keyData in
 		})
 	}
 
+	if importId == "" {
+		return "", diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid import id argument",
+			Detail:   fmt.Sprintf("The import ID value is \"\", please provide a valid value."),
```

Comment:
You don't need the `fmt.Sprintf` here since we're not interpolating anything.

Maybe we could change this to say something like: "The import ID value evaluates to an empty string, please provide a non-empty value". I think that might make it clearer in case people are looking at it and not realising an empty string is coming from several nested/hidden references?